File: readstream.js
let c = 0;
void (url => require('http').get(url,(rdStr, s = 0) => {
rdStr.on('data',d => {
  const v = d.filter(x => x > 0);
  console.log(c + ' ' + (s+=d.byteLength) + ' ' + v);
  if (v == 'cats') process.exit(0);
  ++c;
});
rdStr.on('end',() =>  console.log(`Итог: ${s}`))}))
('http://kodaktor.ru/api2/buffer2/15')

File: readstream.py
from urllib.error import URLError, HTTPError
import urllib.request
import string


def get_stream_data(chunk_size=65536, url="http://kodaktor.ru/api2/buffer2/", N="15"):
    full_url = url + N
    req = urllib.request.Request(full_url)
    counter = 0
    keyword = {}
    try:
        with urllib.request.urlopen(req) as response:
            while True:
                d = response.read(chunk_size)
                counter += chunk_size
                d = d.decode('utf-8')

                for ch in d:
                    if ch in string.ascii_letters:
                        keyword[str(ch)] = str(counter)
                if len(keyword) == 4:
                    break

                print("Chunk size: {}\n".format(counter))

    except (URLError, HTTPError) as e:
        print("Ошибка в запросе к хосту {} ({})".format(full_url, e))

    return keyword


res = get_stream_data()
print(res)
